### ISSUE TITLE
There is an error when I use SpatialDimPlot() and set interactive = TRUE

### ISSUE THREAD

**Issue by yhli1992** [satijalab/seurat#3626]
When I use SpatialDimPlot() and set interactive = TRUE, it shows an error just like this:
` Cannot find 'NA' in this Seurat object`

I use the human breast cancer data from 10x website(https://support.10xgenomics.com/spatial-gene-expression/datasets), and following the workflow by Seurat vignette(https://satijalab.org/seurat/v3.2/spatial_vignette.html). I've tried this workflow under CentOS7 (R 3.6.0 Seurat3.2.2&3.99) and Windows10 (R 4.0.2 Seurat 3.99), and they gave me the same error message.

By the way, when I tried to use the LinkedDimPlot() method, it worked without any problem.

Here is the code what I use: 
```
library(Seurat)
library(ggplot2)

brCancer1 = Load10X_Spatial(
  data.dir = "./data/",
  filename = "V1_Breast_Cancer_Block_A_Section_1_filtered_feature_bc_matrix.h5"
)
brCancer1 = SCTransform(brCancer1,assay = "Spatial")
brCancer1 = RunPCA(brCancer1,assay = "SCT")
brCancer1 = FindNeighbors(brCancer1,dims = 1:30)
brCancer1 = FindClusters(brCancer1)
brCancer1 = RunUMAP(brCancer1,dims = 1:30,min.dist = 0.1,n.neighbors = 50)
SpatialDimPlot(brCancer1,interactive = TRUE)
```
Here is the sessionInfo under CentOS7:
```
> sessionInfo()
R version 3.6.0 (2019-04-26)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] SingleR_1.0.6               SummarizedExperiment_1.16.1 DelayedArray_0.12.3        
 [4] BiocParallel_1.20.1         matrixStats_0.57.0          Biobase_2.46.0             
 [7] GenomicRanges_1.38.0        GenomeInfoDb_1.22.1         IRanges_2.20.2             
[10] S4Vectors_0.24.4            BiocGenerics_0.32.0         shiny_1.5.0                
[13] ggplot2_3.3.2               Seurat_3.9.9.9002          

loaded via a namespace (and not attached):
  [1] Rtsne_0.15                    colorspace_1.4-1              deldir_0.1-29                
  [4] ellipsis_0.3.1                ggridges_0.5.2                XVector_0.26.0               
  [7] BiocNeighbors_1.4.2           rstudioapi_0.11               spatstat.data_1.4-3          
 [10] leiden_0.3.3                  listenv_0.8.0                 farver_2.0.3                 
 [13] bit64_4.0.5                   ggrepel_0.8.2                 AnnotationDbi_1.48.0         
 [16] interactiveDisplayBase_1.24.0 fansi_0.4.1                   codetools_0.2-16             
 [19] splines_3.6.0                 polyclip_1.10-0               jsonlite_1.7.1               
 [22] Cairo_1.5-12.2                ica_1.0-2                     dbplyr_1.4.4                 
 [25] cluster_2.0.8                 png_0.1-7                     uwot_0.1.8.9001              
 [28] sctransform_0.3.1             BiocManager_1.30.10           compiler_3.6.0               
 [31] httr_1.4.2                    assertthat_0.2.1              Matrix_1.2-18                
 [34] fastmap_1.0.1                 lazyeval_0.2.2                cli_2.1.0                    
 [37] later_1.1.0.1                 htmltools_0.5.0               tools_3.6.0                  
 [40] rsvd_1.0.3                    igraph_1.2.6                  gtable_0.3.0                 
 [43] glue_1.4.2                    GenomeInfoDbData_1.2.2        RANN_2.6.1                   
 [46] reshape2_1.4.4                dplyr_1.0.2                   rappdirs_0.3.1               
 [49] Rcpp_1.0.5                    spatstat_1.64-1               vctrs_0.3.4                  
 [52] nlme_3.1-139                  ExperimentHub_1.12.0          DelayedMatrixStats_1.8.0     
 [55] lmtest_0.9-38                 stringr_1.4.0                 globals_0.13.1               
 [58] mime_0.9                      miniUI_0.1.1.1                lifecycle_0.2.0              
 [61] irlba_2.3.3                   goftest_1.2-2                 future_1.19.1                
 [64] AnnotationHub_2.18.0          MASS_7.3-51.4                 zlibbioc_1.32.0              
 [67] zoo_1.8-8                     scales_1.1.1                  promises_1.1.1               
 [70] spatstat.utils_1.17-0         RColorBrewer_1.1-2            yaml_2.2.1                   
 [73] curl_4.3                      memoise_1.1.0                 reticulate_1.16              
 [76] pbapply_1.4-3                 gridExtra_2.3                 rpart_4.1-15                 
 [79] RSQLite_2.2.1                 stringi_1.5.3                 BiocVersion_3.10.1           
 [82] rlang_0.4.8                   pkgconfig_2.0.3               bitops_1.0-6                 
 [85] lattice_0.20-38               ROCR_1.0-11                   purrr_0.3.4                  
 [88] tensor_1.5                    patchwork_1.0.1               htmlwidgets_1.5.2            
 [91] labeling_0.4.2                bit_4.0.4                     cowplot_1.1.0                
 [94] tidyselect_1.1.0              RcppAnnoy_0.0.16              plyr_1.8.6                   
 [97] magrittr_1.5                  R6_2.4.1                      generics_0.0.2               
[100] DBI_1.1.0                     pillar_1.4.6                  withr_2.3.0                  
[103] mgcv_1.8-28                   fitdistrplus_1.1-1            survival_3.2-7               
[106] abind_1.4-5                   RCurl_1.95-4.11               tibble_3.0.4                 
[109] future.apply_1.6.0            crayon_1.3.4                  KernSmooth_2.23-15           
[112] BiocFileCache_1.10.2          plotly_4.9.2.1                grid_3.6.0                   
[115] data.table_1.13.2             blob_1.2.1                    digest_0.6.26                
[118] xtable_1.8-4                  tidyr_1.1.2                   httpuv_1.5.4                 
[121] munsell_0.5.0                 viridisLite_0.3.0    
```

**Comment by zhiiiyang** [comment]
See solution here https://github.com/satijalab/seurat/issues/3216

**Comment by andrewwbutler** [comment]
Hi,

Thanks for reporting. It looks like we could improve the default image handling behavior here. In the meantime, if you explicitly set the `images` parameter, it should enable you to return the plot. E.g.

```R
SpatialDimPlot(brCancer1, interactive = TRUE, images = "slice1")
```

**Comment by andrewwbutler** [comment]
This should now be resolved on the `develop` and `release/4.0.0` branches
